### PR TITLE
Add Jup Fees to the platform

### PR DESCRIPTION
**Pull request by YCrydev**
# Pull Request Description

## Related Issue
Fixes # (issue number)

## Changes Made
This PR adds the following changes:

- Added jupReferralAccount to allow devs to set Fee Accounts on trade API (optional parameter)
- Added jupFeeBps to allow devs to set fee in bps  (optional parameter)
  
## Implementation Details

- Added optional Parameter to trade API

## Transaction executed by agent 

Example transaction: 
https://solscan.io/tx/35b1wz1LjU5pEALq6m5Q6Gyc3QB3ejvsjNAYoy9PdHu9WP8SzGLjHi59sRPj8hvtntVMDMEguWuu273FZ3EbUSoZ
## Prompt Used

```
```

## Checklist
- [x] I have tested these changes locally
- [x] I have updated the documentation
- [x] I have added a transaction link
- [ ] I have added the prompt used to test it

### CODE REVIEW

File: src/agent/index.ts [hunk 1]
```diff
@@ -133,8 +133,10 @@ export class SolanaAgentKit {
     inputAmount: number,
     inputMint?: PublicKey,
     slippageBps: number = DEFAULT_OPTIONS.SLIPPAGE_BPS,
+    jupReferralAccount?:PublicKey,
+    jupFeeBps?:number,
   ): Promise<string> {
-    return trade(this, outputMint, inputAmount, inputMint, slippageBps);
+    return trade(this, outputMint, inputAmount, inputMint, slippageBps, jupReferralAccount ,jupFeeBps);
```
Comment (thearyanag) [inline src/agent/index.ts:139]: instead of passing it as parameter, better to keep it as an optional env parameter?

**Comment by arihantbansal** [comment]
LGTM! Have you tested it out?

File: src/agent/index.ts (hunk 1 above)
Comment (arihantbansal) [reply src/agent/index.ts:139]: Hmm, makes sense, but what if there would be a better way to pass in certain deployment config variables for instantiating an agent?

Comment (thearyanag) [reply src/agent/index.ts:139]: umm, what way can you think of

Comment (arihantbansal) [reply src/agent/index.ts:139]: Maybe a config param, which is an object having parameters defining different config variables, and a default option for them. Might not look as useful right now, but can be helpful down the line when there are more config options needed.

Comment (thearyanag) [reply src/agent/index.ts:139]: Let's start with optional env params, as it would not be a breaking change so can modify it later as need arises

Comment (YCrydev) [reply src/agent/index.ts:139]: Noted will edit!

**Comment by thearyanag** [comment]
Hey @YCrydev looks good now 

can you just run 'pnpm lint:fix' once and push it

**Comment by YCrydev** [comment]
@thearyanag 'pnpm lint:fix' and pushed

**Pull request merged**